### PR TITLE
AGR-2535 add debugging options, add missing filtering options

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

**Comment by cmpich** [comment]
Enable ZFIN variants to be included.

File: agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/GeneRESTInterface.java [hunk 1]
```diff
@@ -121,16 +121,18 @@ Gene getGene(
...
+            @Parameter(in = ParameterIn.QUERY, name = "filter.associatedGeneSymbol", description = "allele symbol", schema = @Schema(type = SchemaType.STRING))
+            @QueryParam("filter.associatedGeneSymbol") String associatedGeneSymbol,
             @Parameter(in = ParameterIn.QUERY, name = "filter.synonym", description = "allele synonym", schema = @Schema(type = SchemaType.STRING))
             @QueryParam("filter.synonym") String synonym,
             @Parameter(in = ParameterIn.QUERY, name = "filter.variantType", description = "Variant types", schema = @Schema(type = SchemaType.STRING))
             @QueryParam("filter.variantType") String variantType,
-            @Parameter(in = ParameterIn.QUERY, name = "filter.variantConsequence", description = "Consequence", schema = @Schema(type = SchemaType.STRING))
-            @QueryParam("filter.variantConsequence") String consequence,
+            @Parameter(in = ParameterIn.QUERY, name = "filter.molecularConsequence", description = "molecularConsequence", schema = @Schema(type = SchemaType.STRING))
+            @QueryParam("filter.molecularConsequence") String consequence,
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/GeneRESTInterface.java:131]: Can we change the variable name (`String consequence`) to match the filter name?

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/GeneRESTInterface.java (hunk 1 above)
Comment (oblodgett) [reply agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/GeneRESTInterface.java:131]: Same with the controller that implements this?

Comment (cmpich) [reply agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/GeneRESTInterface.java:131]: Yes.

File: agr_api/src/main/java/org/alliancegenome/api/controller/AlleleController.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public Allele getAllele(String id) {
         long startTime = System.currentTimeMillis();
         Pagination pagination = new Pagination(page, limit, sortBy, null);
         pagination.addFieldFilter(FieldFilter.VARIANT_TYPE, variantType);
-        pagination.addFieldFilter(FieldFilter.VARIANT_CONSEQUENCE, consequence);
+        pagination.addFieldFilter(FieldFilter.MOLECULAR_CONSEQUENCE, consequence);
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/controller/AlleleController.java:52]: I would guess that `consequence` should be `molecularConsequence`?

File: agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ public Gene getGene(String id) {
         pagination.addFieldFilter(FieldFilter.VARIANT_TYPE, variantType);
         pagination.addFieldFilter(FieldFilter.HAS_DISEASE, hasDisease);
         pagination.addFieldFilter(FieldFilter.HAS_PHENOTYPE, hasPhenotype);
-        pagination.addFieldFilter(FieldFilter.VARIANT_CONSEQUENCE, consequence);
+        pagination.addFieldFilter(FieldFilter.MOLECULAR_CONSEQUENCE, consequence);
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java:93]: I would guess that `consequence` should be `molecularConsequence`?

File: agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java [hunk 4]
```diff
@@ -185,7 +187,7 @@ public Response getAllelesPerGeneDownload(String id,
         pagination.addFieldFilter(FieldFilter.DISEASE, disease);
         pagination.addFieldFilter(FieldFilter.VARIANT_TYPE, variantType);
         pagination.addFieldFilter(FieldFilter.PHENOTYPE, phenotype);
-        pagination.addFieldFilter(FieldFilter.VARIANT_CONSEQUENCE, consequence);
+        pagination.addFieldFilter(FieldFilter.MOLECULAR_CONSEQUENCE, consequence);
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java:190]: I would guess that `consequence` should be `molecularConsequence`?

File: agr_api/src/main/java/org/alliancegenome/api/controller/VariantController.java [hunk 5]
```diff
@@ -35,7 +35,7 @@
         long startTime = System.currentTimeMillis();
         Pagination pagination = new Pagination(page, limit, sortBy, null);
         pagination.addFieldFilter(FieldFilter.VARIANT_TYPE, transcriptType);
-        pagination.addFieldFilter(FieldFilter.VARIANT_CONSEQUENCE, consequence);
+        pagination.addFieldFilter(FieldFilter.MOLECULAR_CONSEQUENCE, consequence);
```
Comment (oblodgett) [inline agr_api/src/main/java/org/alliancegenome/api/controller/VariantController.java:38]: I would guess that `consequence` should be `molecularConsequence`?

File: agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java (hunk 4 above)
Comment (cmpich) [reply agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java]: Yes, let's make it consistent.

File: agr_api/src/main/java/org/alliancegenome/api/controller/VariantController.java (hunk 5 above)
Comment (cmpich) [reply agr_api/src/main/java/org/alliancegenome/api/controller/VariantController.java]: Yes, let's make it consistent.

File: agr_java_core/src/main/java/org/alliancegenome/cache/repository/helper/AlleleFiltering.java [hunk 6]
```diff
@@ -158,7 +158,7 @@ public AlleleFiltering() {
         filterFieldMap.put(FieldFilter.PHENOTYPE, phenotypeFilter);
         filterFieldMap.put(FieldFilter.DISEASE, diseaseFilter);
         filterFieldMap.put(FieldFilter.VARIANT_TYPE, variantTypeFilter);
-        filterFieldMap.put(FieldFilter.VARIANT_CONSEQUENCE, variantConsequenceFilter);
+        filterFieldMap.put(FieldFilter.MOLECULAR_CONSEQUENCE, variantConsequenceFilter);
```
Comment (oblodgett) [inline agr_java_core/src/main/java/org/alliancegenome/cache/repository/helper/AlleleFiltering.java:161]: Sorry stickler for naming, `variantConsequenceFilter` -> `molecularConsequenceFilter`?

And there are a bunch of others.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**